core/common/src/main/scala/net/liftweb/common/Box.scala

```diff
@@ -576,18 +552,18 @@ sealed abstract class Box[+A] extends Product with Serializable{
   def asA[B](implicit m: Manifest[B]): Box[B] = Empty
 
   /**
-   * Return this Box if `Full`, or the specified alternative if it is
-   * empty. Equivalent to `Option`'s `[[scala.Option.orElse orElse]]`.
-   */
-  def or[B >: A](alternative: => Box[B]): Box[B] = alternative
-
-  /**
    * Returns an `[[scala.collection.Iterator Iterator]]` over the value
    * contained in this `Box`, if any.
    */
   def elements: Iterator[A] = Iterator.empty
 
   /**
+    * Return this Box if `Full`, or the specified alternative if it is
+    * empty. Equivalent to `Option`'s `[[scala.Option.orElse orElse]]`.
+    */
+  def or[B >: A](alternative: => Box[B]): Box[B] = transform { case _: EmptyBox => alternative }
```

Comment:
While this is quite clean, I wonder if we should leave `or` as-is. The JVM will be able to turbo-inline the simple definitions of `or`, whereas this definition will probably (though not necessarily) be near-impossible to optimize. Given how common it is to call `or`, I'm tempted to say we should leave it as-is.

Reply:
Hmm. Yeah, I didn't think it along those lines. I'll change that back.

Reply:
Do you think keeping the `or` abstract in the `Box` trait will make much of a difference? I mean, that way, there's one implementation in `Full` and one in `EmptyBox`. It sort of seems cleaner.

Reply:
I think that's fine.